src/app/cli/tool-confirmation/cli-confirmation-handler.ts

```diff
@@ -0,0 +1,229 @@
+import { logger } from '../../../core/logger/index.js';
+import * as readline from 'readline';
+import chalk from 'chalk';
+import { InMemorySettingsProvider } from '../../../core/settings/in-memory-provider.js';
+import { SettingsProvider } from '../../../core/settings/types.js';
+import boxen from 'boxen';
+import {
+    ToolConfirmationEvent,
+    ToolConfirmationResponse,
+} from '../../../core/client/tool-confirmation/types.js';
+import { AgentEventBus } from '../../../core/events/index.js';
+import { EventSubscriber } from '../../api/types.js';
+
+/**
+ * CLI-specific subscriber for tool confirmation events
+ * Implements EventSubscriber pattern to listen to AgentEventBus for tool confirmation requests
+ */
+export class CLIToolConfirmationSubscriber implements EventSubscriber {
+    private settingsProvider: SettingsProvider;
```

Comment:
should we get rid of the `settingsProvider`. not sure what value this has, and it's weird to be using core types in this manner imo